bindings/c/test/apitester/TesterMultiTenantCorrectnessWorkload.cpp

```diff
@@ -0,0 +1,258 @@
+/*
+ * TesterMultiTenantCorrectnessWorkload.cpp
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2013-2022 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "TesterApiWorkload.h"
+#include "TesterUtil.h"
+#include "test/fdb_api.hpp"
+#include <memory>
+#include <map>
+#include <fmt/format.h>
+
+namespace FdbApiTester {
+
+class MultiTenantCorrectnessWorkload : public ApiWorkload {
+public:
+	MultiTenantCorrectnessWorkload(const WorkloadConfig& config) : ApiWorkload(config) {
+		numTenants = config.numTenants;
+	}
+
+private:
+	enum OpType { OP_INSERT, OP_GET, OP_GET_RANGE, OP_CLEAR, OP_CLEAR_RANGE, OP_LAST = OP_CLEAR_RANGE };
+	int numTenants = -1;
+
+	void start() override {
+		schedule([this]() { setupTenants([this]() { runTests(); }, this->numTenants); });
+	}
+
+	void setupTenants(TTaskFct cont, int numTenants) {
+		execTransaction(
+		    [numTenants](auto ctx) {
+			    for (int i = 0; i < numTenants; i++) {
+				    std::string tenant_name = "tenant" + std::to_string(i);
+				    fdb::Tenant::createTenant(ctx->tx(), fdb::toBytesRef(tenant_name));
+			    }
+			    ctx->commit();
+		    },
+		    [this, cont]() { schedule(cont); });
+	}
+
+	void randomGetOp(TTaskFct cont, int tenantId) {
```

Comment:
Implementation of the operations seem to be pretty much a copy-paste from TesterCorrectnessWorkload and TesterApiWorkload. It would then rather make sense to implement it either a configuration option of TesterCorrectnessWorkload